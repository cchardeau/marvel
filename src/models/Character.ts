export type TCharacter = {
  id: number
  name: string
  thumbnail: string
  description: string
  comicsCount: number
  comics: string[]
}
