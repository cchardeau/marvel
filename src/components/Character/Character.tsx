import React from 'react'
import Button from '@material-ui/core/Button'
import Card from '@material-ui/core/Card'
import CardActions from '@material-ui/core/CardActions'
import CardContent from '@material-ui/core/CardContent'
import CardMedia from '@material-ui/core/CardMedia'
import Grid from '@material-ui/core/Grid'
import { makeStyles } from '@material-ui/core/styles'

import CharacterDetails from '../CharacterDetails/CharacterDetails'
import { TCharacter } from '../../models/Character'

type Props = {
  character: TCharacter
}

const useStyles = makeStyles((theme) => ({
  cardGrid: {
    paddingTop: theme.spacing(8),
    paddingBottom: theme.spacing(8)
  },
  card: {
    height: '100%',
    display: 'flex',
    flexDirection: 'column'
  },
  cardMedia: {
    paddingTop: '56.25%' // 16:9
  },
  cardContent: {
    flexGrow: 1
  }
}))

const Character: React.FC<Props> = ({ character }) => {
  // styling
  const classes = useStyles()

  return (
    <React.Fragment key={character.id}>
      <Grid item xs={12} sm={6} md={4}>
        <Card className={classes.card}>
          <CardMedia
            className={classes.cardMedia}
            image={character.thumbnail}
            title={`${character.name} image`}
          />
          <CardContent className={classes.cardContent}>
            <CharacterDetails character={character} />
          </CardContent>
          <CardActions>
            <Button size="small" color="primary">
              Set as favorite
            </Button>
          </CardActions>
        </Card>
      </Grid>
    </React.Fragment>
  )
}

export default Character