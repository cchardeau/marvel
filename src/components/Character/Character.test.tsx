import React from 'react'
import ReactDOM from 'react-dom'
import renderer from 'react-test-renderer'

import Character from './Character'
import { TCharacter } from '../../models/Character'

const mockCharacter: TCharacter = {
  id: 1,
  name: 'name',
  thumbnail: 'https://thumbnail',
  description: 'description', 
  comicsCount: 42,
  comics: []
}

it('renders without crashing', () => {
  const div = document.createElement('div')
  ReactDOM.render(<Character character={mockCharacter} />, div)
  ReactDOM.unmountComponentAtNode(div)
})

it('renders correctly', () => {
  const output = renderer.create(<Character character={mockCharacter} />).toJSON()
  expect(output).toMatchSnapshot()
})
