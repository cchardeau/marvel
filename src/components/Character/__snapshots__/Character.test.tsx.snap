// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6 MuiGrid-grid-md-4"
>
  <div
    className="MuiPaper-root MuiPaper-elevation1 MuiCard-root makeStyles-card-221 MuiPaper-rounded"
  >
    <div
      className="MuiCardMedia-root makeStyles-cardMedia-222"
      style={
        Object {
          "backgroundImage": "url(\\"https://thumbnail\\")",
        }
      }
      title="name image"
    />
    <div
      className="MuiCardContent-root makeStyles-cardContent-223"
    >
      <h2
        className="MuiTypography-root makeStyles-characterName-359 MuiTypography-h5 MuiTypography-gutterBottom"
        onClick={[Function]}
      >
        name
      </h2>
    </div>
    <div
      className="MuiCardActions-root MuiCardActions-spacing"
    >
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-textSizeSmall MuiButton-sizeSmall"
        disabled={false}
        onBlur={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiButton-label"
        >
          Set as favorite
        </span>
      </button>
    </div>
  </div>
</div>
`;
